src/preferences/effectchainpresetlistmodel.cpp

```diff
@@ -0,0 +1,71 @@
+#include "preferences/effectchainpresetlistmodel.h"
+
+#include <QMimeData>
+
+namespace {
+const QString kMimeTextDelimeter = "\n";
+} // anonymous namespace
+
+QMimeData* EffectChainPresetListModel::mimeData(const QModelIndexList& indexes) const {
+    QMimeData* mimeData = new QMimeData();
+    QStringList chainNameList;
+    for (const auto& index : indexes) {
+        if (index.isValid()) {
+            chainNameList << data(index).toString();
+        }
+    }
+    mimeData->setText(chainNameList.join(kMimeTextDelimeter));
+    return mimeData;
+}
+
+bool EffectChainPresetListModel::dropMimeData(
+        const QMimeData* data,
+        Qt::DropAction action,
+        int row,
+        int column,
+        const QModelIndex& parent) {
+    Q_UNUSED(column);
+    Q_UNUSED(action);
+    if (!data->hasText()) {
+        return false;
+    }
+    if (row == -1) {
+        row = parent.row();
+        // Dropping onto an empty model or dropping past the end of a model
+        if (parent.row() == -1) {
+            row = stringList().size();
+        }
+    }
+    QStringList mimeTextLines = data->text().split(kMimeTextDelimeter);
```

Comment:
```suggestion
    const QStringList mimeTextLines = data->text().split(kMimeTextDelimeter);
```
or use qAsConst in the for loop to prevent detaching